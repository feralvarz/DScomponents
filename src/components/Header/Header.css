.ds-header
  .p-menubar
  .p-menubar-root-list
  > .p-menuitem
  > .p-menuitem-link
  :is(.p-menuitem-text, .p-submenu-icon) {
  @apply text-ds-base hover:text-white bg-transparent;
}

.ds-header-start > * {
  @apply max-h-[48px];
}

.ds-header .p-menubar .p-menubar-root-list .p-menuitem-link {
  @apply font-semi min-h-[48px] !text-ds-base hover:!text-white !bg-transparent;
}

.ds-header .p-menubar .p-menubar-root-list > .p-menuitem > .p-menuitem-link:focus {
  outline: 0 none;
  outline-offset: 0;
  box-shadow: none;
}

.p-menubar .p-menubar-root-list > .p-menuitem:is(:last-child) > .p-menuitem-link svg {
  @apply hidden;
}

.ds-header .p-menubar .p-menubar-root-list > .p-menuitem > .p-menuitem-link:not(.p-disabled):hover {
  @apply bg-transparent;
}

.ds-header .p-menubar .p-menuitem-text {
  @apply text-inherit;
}

.ds-header .p-submenu-list {
  @apply bg-primary right-0 min-w-[16rem];
}

.ds-header
  .p-menubar
  .p-menubar-root-list
  .p-submenu-list
  .p-menuitem.p-menuitem-active
  .p-menuitem-link:not(.p-disabled):hover {
  @apply !bg-primary-950;
}

.ds-header .p-menubar .p-menu-separator {
  @apply border-ds-neutral;
}

.ds-header .p-menubar > .p-menuitem-link:not(.p-disabled):hover {
  @apply text-white;
}

.ds-header .p-menubar .p-menuitem.p-menuitem-active > .p-menuitem-link .p-menuitem-text {
  @apply !text-white hover:!text-white;
}

.ds-header
  .p-menubar
  .p-menubar-root-list
  > .p-menuitem
  > .p-menuitem-link:not(.p-disabled):hover
  .p-menuitem-text {
  @apply text-white;
}
