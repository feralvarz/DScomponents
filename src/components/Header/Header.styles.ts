import ctl from '@netlify/classnames-template-literals';
import { twMerge } from 'tailwind-merge';

const baseStyles = ctl(`
  bg-primary-900
  text-base
  rounded-none
  border-primary-900
  flex
  flex-row
  justify-between
  text-white
  px-6
  py-2
`);

export const getClassName = (className = '') => twMerge(baseStyles, className);
