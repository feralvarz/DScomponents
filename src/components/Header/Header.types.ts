import type { MenuItem } from 'primereact/menuitem';
import { MenubarProps } from 'primereact/menubar';

export type HeaderProps = {
  startItem: React.ReactNode;
  endItem?: React.ReactNode;
} & MenubarProps;
