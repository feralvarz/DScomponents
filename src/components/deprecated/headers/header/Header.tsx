import { Menubar } from 'primereact/menubar';
import { MenuItem } from 'primereact/menuitem';
import type { DSComponentProps } from 'src/types';
import type { FC } from 'react';

interface HeaderProps extends DSComponentProps {
  items?: MenuItem[];
  startItem?: React.ReactNode;
  endItem?: React.ReactNode;
  justify?: 'start' | 'end' | 'center' | 'between' | 'around';
  className?: string;
}

const twJustifyClass = {
  center: 'justify-center',
  start: 'justify-start',
  end: 'justify-end',
  between: 'justify-between',
  around: 'justify-around',
};

/**
 * @deprecated use Header from components/Header/Header
 */
const Header: FC<HeaderProps> = ({
  startItem,
  endItem,
  theme = 'main',
  items = [],
  justify = 'between',
  className,
}) => {
  return (
    <header className={className} data-theme={theme}>
      <Menubar
        className={`bg-primary text-base-content rounded-none border-primary flex flex-row ${twJustifyClass[justify]}`}
        end={endItem}
        model={items}
        start={startItem}
      />
    </header>
  );
};

export default Header;
