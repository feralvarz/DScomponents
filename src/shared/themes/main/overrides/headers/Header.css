.p-menubar {
  @apply bg-primary-900;
}

.p-menubar .p-menubar-root-list > .p-menuitem > .p-menuitem-link .p-menuitem-text {
  @apply text-ds-base;
}

.p-menubar .p-menubar-root-list > .p-menuitem > .p-menuitem-link .p-submenu-icon {
  @apply text-ds-base;
}

.p-menubar .p-menubar-root-list > .p-menuitem > .p-menuitem-link:focus {
  outline: 0 none;
  outline-offset: 0;
  box-shadow: none;
}

.p-menubar.p-menubar-mobile-active .p-menubar-root-list {
  @apply bg-primary-900;
}

.p-menubar .p-menubar-root-list > .p-menuitem.p-menuitem-active > .p-menuitem-link,
.p-menubar
  .p-menubar-root-list
  > .p-menuitem.p-menuitem-active
  > .p-menuitem-link:not(.p-disabled):hover {
  @apply bg-primary-900;
}

.p-menubar .p-menubar-root-list > .p-menuitem > .p-menuitem-link:not(.p-disabled):hover {
  @apply bg-primary-900;
}

.p-menubar-root-list > .p-menuitem-active > .p-submenu-list {
  @apply bg-primary-900;
}

.p-menubar .p-menuitem-link .p-menuitem-text {
  @apply text-ds-base;
}

.p-menubar .p-menuitem-link:not(.p-disabled):hover {
  @apply bg-primary-900;
}

.p-menubar .p-menuitem-link:not(.p-disabled):hover .p-menuitem-text {
  @apply text-ds-base;
}

.p-menubar .p-menubar-root-list .p-menu-separator {
  @apply border-ds-neutral;
}

.p-menubar .p-menubar-button:hover {
  @apply bg-primary-900;
}

.p-menubar .p-menubar-button:focus {
  @apply shadow-md;
}

.p-menubar .p-menubar-root-list > .p-menuitem.p-menuitem-active > .p-menuitem-link .p-menuitem-text,
.p-menubar
  .p-menubar-root-list
  > .p-menuitem.p-menuitem-active
  > .p-menuitem-link:not(.p-disabled):hover
  .p-menuitem-text {
  @apply text-ds-base;
}

.p-menubar .p-menuitem.p-menuitem-active > .p-menuitem-link {
  @apply bg-primary-900;
}

.p-menubar
  .p-menubar-root-list
  > .p-menuitem
  > .p-menuitem-link:not(.p-disabled):hover
  .p-menuitem-text {
  @apply text-ds-base;
}

.p-menubar
  .p-menubar-root-list
  > .p-menuitem
  > .p-menuitem-link:not(.p-disabled):hover
  .p-submenu-icon {
  @apply text-ds-base;
}

.p-menubar .p-menubar-root-list > .p-menuitem.p-menuitem-active > .p-menuitem-link .p-submenu-icon,
.p-menubar
  .p-menubar-root-list
  > .p-menuitem.p-menuitem-active
  > .p-menuitem-link:not(.p-disabled):hover
  .p-submenu-icon {
  @apply text-ds-base;
}

.p-menubar .p-menuitem.p-menuitem-active > .p-menuitem-link .p-menuitem-text {
  @apply text-ds-base;
}

.p-menubar .p-menuitem-link {
  @apply text-ds-base;
}

.p-menubar .p-submenu-list {
  right: 0;
  min-width: 15rem;
}
