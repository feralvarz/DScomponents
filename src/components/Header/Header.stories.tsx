import logo from '../../assets/logo-ityneri.svg';

import { Header } from './Header';
import type { MenuItem } from 'primereact/menuitem';

const model: MenuItem[] = [
  {
    label: 'Red seleccionada',
    items: [
      {
        label: 'Red Actioticket',
        template: (item: MenuItem) => (
          <a
            href="#"
            role="menuitem"
            className="p-menuitem-link"
            aria-haspopup="false"
            onClick={(e) => {
              e.preventDefault();
              console.log('Click on item', item);
            }}
          >
            <span className="p-menuitem-text">{item.label}</span>
          </a>
        ),
      },
      { label: 'Item Two' },
    ],
  },
  {
    icon: <i className="ri-user-settings-line text-white text-h3" />,
    items: [
      { label: 'admin_ccw@yopmail.com', disabled: true },
      { separator: true },
      { label: 'Cambiar contraseña' },
      { label: 'Cambiar idioma' },
      { label: 'Cerrar sesión' },
    ],
  },
];

const design = {
  type: 'figma',
  url: 'https://www.figma.com/file/lqwCGcf4nD8K5x5HnZrNu3/Geoactio---Design-System?type=design&node-id=80%3A7133&mode=design&t=rK4E9ztEdAdaq7KU-1',
};

export default {
  title: 'Design System/Menus/Header',
  component: Header,
};

export const Default = {
  title: 'Default',
  args: {
    startItem: <img alt="Logo" src={logo} />,
    model: [model[1]],
  },
  parameters: {
    design,
  },
};

export const WithItems = {
  title: 'With items',
  args: {
    startItem: <img alt="Logo" src={logo} />,
    model,
  },
  parameters: {
    design,
  },
};
