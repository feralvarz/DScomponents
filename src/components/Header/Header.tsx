import './Header.css';
import { memo } from 'react';
import type { HeaderProps } from './Header.types';

import { Menubar } from 'primereact/menubar';
import { getClassName } from './Header.styles';

/**
 * Header component
 */
export const Header = memo(function Header(props: HeaderProps) {
  const { model = [], startItem, endItem } = props;

  return (
    <header className="ds-header">
      <Menubar
        className={getClassName(props?.className)}
        start={startItem}
        end={endItem}
        model={model}
        pt={{
          start: () => ({
            className: 'ds-header-start',
          }),
        }}
      />
    </header>
  );
});
