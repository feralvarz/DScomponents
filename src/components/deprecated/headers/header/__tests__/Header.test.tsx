import { render } from '@testing-library/react';
import Header from '../Header';

describe('Header', () => {
  it('renders', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it('renders with props', () => {
    const { container } = render(
      <Header
        endItem={<div>end</div>}
        items={[
          {
            label: 'File',
            icon: 'pi pi-fw pi-file',
            items: [
              {
                label: 'New',
                icon: 'pi pi-fw pi-plus',
                items: [
                  { label: 'Bookmark', icon: 'pi pi-fw pi-bookmark' },
                  { label: 'Video', icon: 'pi pi-fw pi-video' },
                ],
              },
              { label: 'Delete', icon: 'pi pi-fw pi-trash' },
              { separator: true },
              { label: 'Export', icon: 'pi pi-fw pi-external-link' },
            ],
          },
          {
            label: 'Edit',
            icon: 'pi pi-fw pi-pencil',
            items: [
              { label: 'Left', icon: 'pi pi-fw pi-align-left' },
              { label: 'Right', icon: 'pi pi-fw pi-align-right' },
              { label: 'Center', icon: 'pi pi-fw pi-align-center' },
              { label: 'Justify', icon: 'pi pi-fw pi-align-justify' },
            ],
          },
          {
            label: 'Users',
            icon: 'pi pi-fw pi-user',
            items: [
              { label: 'New', icon: 'pi pi-fw pi-user-plus' },
              { label: 'Delete', icon: 'pi pi-fw pi-user-minus' },
              {
                label: 'Search',
                icon: 'pi pi-fw pi-users',
                items: [
                  {
                    label: 'Filter',
                    icon: 'pi pi-fw pi-filter',
                    items: [{ label: 'Print', icon: 'pi pi-fw pi-print' }],
                  },
                ],
              },
            ],
          },
          {
            label: 'Events',
            icon: 'pi pi-fw pi-calendar',
            items: [
              {
                label: 'Edit',
                icon: 'pi pi-fw pi-pencil',
                items: [{ label: 'Save', icon: 'pi pi-fw pi-calendar-plus' }],
              },
              { label: 'Archieve', icon: 'pi pi-fw pi-calendar-minus' },
            ],
          },
        ]}
        justify="between"
        startItem={<div>start</div>}
        theme="main"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
