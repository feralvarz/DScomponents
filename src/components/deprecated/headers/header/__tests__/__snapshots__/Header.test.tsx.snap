// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div>
  <header
    data-theme="main"
  >
    <div
      class="p-menubar p-component bg-primary text-base-content rounded-none border-primary flex flex-row justify-between"
    >
      <ul
        class="p-menubar-root-list"
        role="menubar"
      />
    </div>
  </header>
</div>
`;

exports[`Header renders with props 1`] = `
<div>
  <header
    data-theme="main"
  >
    <div
      class="p-menubar p-component bg-primary text-base-content rounded-none border-primary flex flex-row justify-between"
    >
      <div
        class="p-menubar-start"
      >
        <div>
          start
        </div>
      </div>
      <a
        class="p-menubar-button"
        href="#"
        role="button"
        tabindex="0"
      >
        <i
          class="pi pi-bars"
        />
      </a>
      <ul
        class="p-menubar-root-list"
        role="menubar"
      >
        <li
          class="p-menuitem"
          role="none"
        >
          <a
            aria-haspopup="true"
            class="p-menuitem-link"
            href="#"
            role="menuitem"
          >
            <span
              class="p-menuitem-icon pi pi-fw pi-file"
            />
            <span
              class="p-menuitem-text"
            >
              File
            </span>
            <span
              class="p-submenu-icon pi pi-angle-down"
            />
          </a>
          <ul
            class="p-submenu-list"
            role="menu"
          >
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="true"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-plus"
                />
                <span
                  class="p-menuitem-text"
                >
                  New
                </span>
                <span
                  class="p-submenu-icon pi pi-angle-right"
                />
              </a>
              <ul
                class="p-submenu-list"
                role="menu"
              >
                <li
                  class="p-menuitem"
                  role="none"
                >
                  <a
                    aria-haspopup="false"
                    class="p-menuitem-link"
                    href="#"
                    role="menuitem"
                  >
                    <span
                      class="p-menuitem-icon pi pi-fw pi-bookmark"
                    />
                    <span
                      class="p-menuitem-text"
                    >
                      Bookmark
                    </span>
                  </a>
                </li>
                <li
                  class="p-menuitem"
                  role="none"
                >
                  <a
                    aria-haspopup="false"
                    class="p-menuitem-link"
                    href="#"
                    role="menuitem"
                  >
                    <span
                      class="p-menuitem-icon pi pi-fw pi-video"
                    />
                    <span
                      class="p-menuitem-text"
                    >
                      Video
                    </span>
                  </a>
                </li>
              </ul>
            </li>
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="false"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-trash"
                />
                <span
                  class="p-menuitem-text"
                >
                  Delete
                </span>
              </a>
            </li>
            <li
              class="p-menu-separator"
              role="separator"
            />
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="false"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-external-link"
                />
                <span
                  class="p-menuitem-text"
                >
                  Export
                </span>
              </a>
            </li>
          </ul>
        </li>
        <li
          class="p-menuitem"
          role="none"
        >
          <a
            aria-haspopup="true"
            class="p-menuitem-link"
            href="#"
            role="menuitem"
          >
            <span
              class="p-menuitem-icon pi pi-fw pi-pencil"
            />
            <span
              class="p-menuitem-text"
            >
              Edit
            </span>
            <span
              class="p-submenu-icon pi pi-angle-down"
            />
          </a>
          <ul
            class="p-submenu-list"
            role="menu"
          >
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="false"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-align-left"
                />
                <span
                  class="p-menuitem-text"
                >
                  Left
                </span>
              </a>
            </li>
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="false"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-align-right"
                />
                <span
                  class="p-menuitem-text"
                >
                  Right
                </span>
              </a>
            </li>
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="false"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-align-center"
                />
                <span
                  class="p-menuitem-text"
                >
                  Center
                </span>
              </a>
            </li>
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="false"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-align-justify"
                />
                <span
                  class="p-menuitem-text"
                >
                  Justify
                </span>
              </a>
            </li>
          </ul>
        </li>
        <li
          class="p-menuitem"
          role="none"
        >
          <a
            aria-haspopup="true"
            class="p-menuitem-link"
            href="#"
            role="menuitem"
          >
            <span
              class="p-menuitem-icon pi pi-fw pi-user"
            />
            <span
              class="p-menuitem-text"
            >
              Users
            </span>
            <span
              class="p-submenu-icon pi pi-angle-down"
            />
          </a>
          <ul
            class="p-submenu-list"
            role="menu"
          >
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="false"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-user-plus"
                />
                <span
                  class="p-menuitem-text"
                >
                  New
                </span>
              </a>
            </li>
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="false"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-user-minus"
                />
                <span
                  class="p-menuitem-text"
                >
                  Delete
                </span>
              </a>
            </li>
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="true"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-users"
                />
                <span
                  class="p-menuitem-text"
                >
                  Search
                </span>
                <span
                  class="p-submenu-icon pi pi-angle-right"
                />
              </a>
              <ul
                class="p-submenu-list"
                role="menu"
              >
                <li
                  class="p-menuitem"
                  role="none"
                >
                  <a
                    aria-haspopup="true"
                    class="p-menuitem-link"
                    href="#"
                    role="menuitem"
                  >
                    <span
                      class="p-menuitem-icon pi pi-fw pi-filter"
                    />
                    <span
                      class="p-menuitem-text"
                    >
                      Filter
                    </span>
                    <span
                      class="p-submenu-icon pi pi-angle-right"
                    />
                  </a>
                  <ul
                    class="p-submenu-list"
                    role="menu"
                  >
                    <li
                      class="p-menuitem"
                      role="none"
                    >
                      <a
                        aria-haspopup="false"
                        class="p-menuitem-link"
                        href="#"
                        role="menuitem"
                      >
                        <span
                          class="p-menuitem-icon pi pi-fw pi-print"
                        />
                        <span
                          class="p-menuitem-text"
                        >
                          Print
                        </span>
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li
          class="p-menuitem"
          role="none"
        >
          <a
            aria-haspopup="true"
            class="p-menuitem-link"
            href="#"
            role="menuitem"
          >
            <span
              class="p-menuitem-icon pi pi-fw pi-calendar"
            />
            <span
              class="p-menuitem-text"
            >
              Events
            </span>
            <span
              class="p-submenu-icon pi pi-angle-down"
            />
          </a>
          <ul
            class="p-submenu-list"
            role="menu"
          >
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="true"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-pencil"
                />
                <span
                  class="p-menuitem-text"
                >
                  Edit
                </span>
                <span
                  class="p-submenu-icon pi pi-angle-right"
                />
              </a>
              <ul
                class="p-submenu-list"
                role="menu"
              >
                <li
                  class="p-menuitem"
                  role="none"
                >
                  <a
                    aria-haspopup="false"
                    class="p-menuitem-link"
                    href="#"
                    role="menuitem"
                  >
                    <span
                      class="p-menuitem-icon pi pi-fw pi-calendar-plus"
                    />
                    <span
                      class="p-menuitem-text"
                    >
                      Save
                    </span>
                  </a>
                </li>
              </ul>
            </li>
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                aria-haspopup="false"
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="p-menuitem-icon pi pi-fw pi-calendar-minus"
                />
                <span
                  class="p-menuitem-text"
                >
                  Archieve
                </span>
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <div
        class="p-menubar-end"
      >
        <div>
          end
        </div>
      </div>
    </div>
  </header>
</div>
`;
